rfcs/text/0030/user.yml

```diff
@@ -19,18 +19,9 @@
   reusable:
     top_level: true # the top level user relates to the top level process
     expected:
-      - at: process.entry_leader
+      - at: process
```

Comment:
same issue as with group.

Reply:
We not going to track the real, effective & saved uids & usernames of all processes in the event (leaders etc), just the .process one now?

Reply:
we do by the fact that user can be nested under process. As far as I'm aware, this means the leader processes will have this too? @kgeller let me know if that isn't correct.

Reply:
Yes, the leader processes will get the user and user.group fieldsets. 

The caveat to that, like I mention above, is, as of right now, it won't get the user.effective, user.saved and user.real.

Reply:
hmm, ok so the double nestings won't make the trip.. which means we can't have   process.entry_leader.user.real.name  etc...  cc @m-sample... process.entry_leader.user and process.entry_leader.group makes the trip, but due to the tooling, the self nestings are evaluated after the foreign ones (explained by @kgeller).. 

Do you think this is an issue Mike? We will have the rich nestings for user/group on the top level process, but perhaps fine if they don't get set for leaders?

Reply:
The Linux process model has real, effective, saved (and file which we leave out) *per* process so tracking them for each process referenced in an event is valuable (e.g. see if privileges have increased via a setuid exec or dropped by setuid, etc system calls.). Note for future: we should consider adding the capability set of a process so we can see when privileges change

For now it seems we don't have query a language that can do cross field comparison (e.g. process.session_leader.effective.uid != process.effective.uid).   We have resorted to computed synthetic fields like same_as_process to make some of these comparisons queriable.  Do we need a synthetic field such as "same_users_as_process" for these too?

Reply:
after a few discussions we are removing the self nesting for user and group and breaking out into individual bespoke nestings under process.